web_src/js/components/RepoBranchTagSelector.vue

```diff
@@ -246,7 +246,35 @@ const sfc = {
         event.preventDefault();
         this.menuVisible = false;
       }
-    }
+    },
+    handleTabSwitch(mode) {
+      this.createTag = mode === 'tags';
+      this.mode = mode;
+      this.focusSearchField();
+      this.fetchBranchesOrTags();
+    },
+    async fetchBranchesOrTags() {
+      if (!['branches', 'tags'].includes(this.mode) || this.isLoading) return;
+      // only fetch when branch/tag list has not been initialized
+      if (this.hasListInitialized[this.mode] ||
+      (this.mode === 'branches' && !this.showBranchesInDropdown) || (this.mode === 'tags' && this.noTag)) return;
+      this.isLoading = true;
+      this.hasListInitialized[this.mode] = true;
+      // the "data.defaultBranch" is ambiguous, it could be "branch name" or "tag name"
+      const reqUrl = `${this.repoLink}/${this.mode}/list`;
+      const resp = await fetch(reqUrl);
+      const {results} = await resp.json();
+      for (const result of results) {
+        let selected = false;
+        if (this.mode === 'branches') {
+          selected = result === this.defaultBranch;
+        } else {
+          selected = result === (this.release ? this.release.tagName : this.defaultBranch);
+        }
+        this.items.push({name: result, url: result, branch: this.mode === 'branches', tag: this.mode === 'tags', selected});
```

Comment:
It's not right here.

You are accessing an undetermined variable in an async function. The "this.mode" might not be the mode for the "results"

Reply:
Added condition `if (this.isLoading) return; ` to `handleTabSwitch ` to avoid this
https://github.com/go-gitea/gitea/pull/25719/files#diff-588b4b6c148c056a96c4cb1732bb5a12b12ee31ed2e2938eaf751b44e5f9fd64